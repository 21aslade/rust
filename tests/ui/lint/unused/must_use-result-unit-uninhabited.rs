//@ edition: 2024
//@ aux-crate:dep=must_use_result_unit_uninhabited_extern_crate.rs

#![deny(unused_must_use)]
#![feature(never_type)]

use dep::{MyUninhabited, MyUninhabitedNonexhaustive};

fn f1() -> Result<(), ()> {
    Ok(())
}

fn f2() -> Result<(), core::convert::Infallible> {
    Ok(())
}

fn f3() -> Result<(), !> {
    Ok(())
}

fn f4() -> Result<(), MyUninhabited> {
    Ok(())
}

fn f5() -> Result<(), MyUninhabitedNonexhaustive> {
    Ok(())
}

trait AssocType {
    type Error;
}

struct S1;
impl AssocType for S1 {
    type Error = !;
}

struct S2;
impl AssocType for S2 {
    type Error = ();
}

fn f6<AT: AssocType>(_: AT) -> Result<(), AT::Error> {
    Ok(())
}

fn main() {
    f1(); //~ ERROR: unused `Result` that must be used
    f2();
    f3();
    f4();
    f5(); //~ ERROR: unused `Result` that must be used
    f6(S1);
    f6(S2); //~ ERROR: unused `Result` that must be used
}
