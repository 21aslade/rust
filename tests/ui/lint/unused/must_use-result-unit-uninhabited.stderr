error: unused `Result` that must be used
  --> $DIR/must_use-result-unit-uninhabited.rs:48:5
   |
LL |     f1();
   |     ^^^^
   |
   = note: this `Result` may be an `Err` variant, which should be handled
note: the lint level is defined here
  --> $DIR/must_use-result-unit-uninhabited.rs:4:9
   |
LL | #![deny(unused_must_use)]
   |         ^^^^^^^^^^^^^^^
help: use `let _ = ...` to ignore the resulting value
   |
LL |     let _ = f1();
   |     +++++++

error: unused `Result` that must be used
  --> $DIR/must_use-result-unit-uninhabited.rs:52:5
   |
LL |     f5();
   |     ^^^^
   |
   = note: this `Result` may be an `Err` variant, which should be handled
help: use `let _ = ...` to ignore the resulting value
   |
LL |     let _ = f5();
   |     +++++++

error: unused `Result` that must be used
  --> $DIR/must_use-result-unit-uninhabited.rs:54:5
   |
LL |     f6(S2);
   |     ^^^^^^
   |
   = note: this `Result` may be an `Err` variant, which should be handled
help: use `let _ = ...` to ignore the resulting value
   |
LL |     let _ = f6(S2);
   |     +++++++

error: aborting due to 3 previous errors

